src/consumer/component-ops/add-components/add-components.js

```diff
@@ -114,21 +115,51 @@ export default class AddComponents {
   ignoreList: string[];
   gitIgnore: any;
   origin: ComponentOrigin;
+  configuredConsumer: boolean;
   constructor(consumer: Consumer, addProps: AddProps) {
+    this.configuredConsumer = addProps.configuredConsumer;
+    const consumerPath = consumer.getPath();
     this.consumer = consumer;
     this.bitMap = consumer.bitMap;
-    this.componentPaths = addProps.componentPaths;
+    this.componentPaths =
+      this.configuredConsumer === true
+        ? addProps.componentPaths.map(file => path.join(consumerPath, file))
+        : addProps.componentPaths;
     this.id = addProps.id;
     this.main = addProps.main;
     this.namespace = addProps.namespace;
-    this.tests = addProps.tests || [];
-    this.exclude = addProps.exclude || [];
+    this.initializeTests(addProps);
+    this.initializeExcluded(addProps);
     this.override = addProps.override;
     this.trackDirFeature = addProps.trackDirFeature;
     this.origin = addProps.origin || COMPONENT_ORIGINS.AUTHORED;
     this.warnings = {};
   }
 
+  initializeExcluded(addProps: AddProps) {
+    if (addProps.exclude && addProps.exclude.length) {
+      if (addProps.configuredConsumer === true) {
+        this.exclude = addProps.exclude.map(file => path.join(this.consumer.getPath(), file));
+      } else {
+        this.exclude = addProps.exclude;
+      }
+    } else {
+      this.exclude = [];
+    }
+  }
+
+  initializeTests(addProps: AddProps) {
```

Comment:
this is wet (Write Every Time) it's exactly the same as initializeExcluded, extract the logic to one function